src/Features/Core/Portable/InitializeParameter/AbstractInitializeParameterCodeRefactoringProvider.cs

```diff
@@ -41,69 +46,108 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
             var position = context.Span.Start;
             var document = context.Document;
             var cancellationToken = context.CancellationToken;
-
             var syntaxTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
             var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-
             var token = root.FindToken(position);
-            var parameterNode = GetParameterNode(token, position);
-            if (parameterNode == null)
-            {
-                return;
-            }
+            var firstparameterNode = GetParameterNode(token, position);
+
+            // Creating a list and iterating it from the allows the right order or ifs
+            var listOfParameterNodes = GetParameterList(firstparameterNode);
+            var counter = 0;
 
-            // Only offered when there isn't a selection, or the selection exactly selects
-            // a parameter name.
-            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
-            if (!context.Span.IsEmpty)
+            // List with parameterNodes that pass all checks
+            var listOfValidParametersNodes = new List<SyntaxNode>();
+            var checkIfToReturn = false;
+
+            if (listOfParameterNodes != null)
```

Comment:
This would break existing features that derive from `AbstractInitializeParameterCodeRefactoringProvider`

Reply:
Also, it might be a good idea to consider the possibility of supporting "add for all paramerters" option for them as well, when you fixing it.

---
In reply to: [300800764](https://github.com/dotnet/roslyn/pull/36931#discussion_r300800764) [](ancestors = 300800764)